images/node/scripts/origin-node-run.sh

```diff
@@ -2,21 +2,16 @@
 
 set -eu
 
+hostetc=${HOST_ETC:-/rootfs/etc}
 conf=${CONFIG_FILE:-/etc/origin/node/node-config.yaml}
 opts=${OPTIONS:---loglevel=2}
-
-function quit {
-    pkill -g 0 openshift
-    exit 0
-}
-
-trap quit SIGTERM
-
-if [ ! -f ${HOST_ETC}/systemd/system/docker.service.d/docker-sdn-ovs.conf ]; then
-    mkdir -p ${HOST_ETC}/systemd/system/docker.service.d
-    cp /usr/lib/systemd/system/docker.service.d/docker-sdn-ovs.conf ${HOST_ETC}/systemd/system/docker.service.d
+if [ "$#" -ne 0 ]; then
+  opts=""
 fi
 
-/usr/bin/openshift start node "--config=${conf}" "${opts}" &
+if [ ! -f ${hostetc}/systemd/system/docker.service.d/docker-sdn-ovs.conf ]; then
+    mkdir -p ${hostetc}/systemd/system/docker.service.d
+    cp /usr/lib/systemd/system/docker.service.d/docker-sdn-ovs.conf ${hostetc}/systemd/system/docker.service.d
+fi
 
-while true; do sleep 5; done
+exec /usr/bin/openshift start node "--config=${conf}" "${opts}" $@
```

Comment:
Yes, if you don't exec then signals aren't sent to the process.  Trapping
is only necessary if you want to do other process management.

On Wed, Apr 27, 2016 at 2:12 PM, Scott Dodson <notifications@github.com>
wrote:

> In images/node/scripts/origin-node-run.sh
> <https://github.com/openshift/origin/pull/7284#discussion_r61308530>:
>
> >
> > -while true; do sleep 5; done
> > +exec /usr/bin/openshift start node "--config=${conf}" "${opts}" $@
>
> Does exec behave differently than just calling the binary? I believe I ran
> into problems where the signals weren't being sent to the openshift process
> without the trapping and killing.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7284/files/da059644a58ca77319f89af0afea44fa4ab17302#r61308530>
>